build/helper/metadata_add_all.py

```diff
@@ -440,57 +440,103 @@ def add_all_attribute_metadata(attributes, config):
 
 
 def _add_enum_codegen_method(enums, config):
-    '''Adds 'codegen_method' that will determine whether and how the enum is code-generated.
+    '''Adds 'codegen_method' if not explicitly specified that will determine whether and how the enum is code-generated.
 
-    Set the 'codegen_method' of all enums that do not explicitly specify it in metadata to 'no',
-    then go through all functions and attributes and set to the least restrictive use.
-    If an enum without converter has an explicit non-public 'codegen_method' but it is used by a
-    'public'/'python-only' function or 'public' attribute, a ValueError would be thrown.
+    If an enum does not explicitly specify its 'codegen_method' in metadata, it will be assigned the
+    least restrictive codegen_method based on the codegen_method of all the functions and attributes
+    that use it. The default codegen_method is 'no' (if no function nor attribute uses it).
+
+    If an enum explicitly specifies its 'codegen_method' in metadata, it will be checked against the
+    least restrictive codegen_method based on the codegen_method of all the functions and attributes
+    that use it (if its explicit 'codegen_method' is more restrictive than the calculated least
+    restrictive codegen_method and it does not use converter, a ValueError would be thrown).
+
+    The restrictiveness of the codegen_method is as follows (most restrictive -> least restrictive):
+    'no' -> 'private' -> 'public' / 'python-only' (will be converted to 'public' if not explicitly specified)
     '''
-    enum_has_explicit_codegen_method = {}
+    enum_to_client_functions = _get_functions_that_use_enums(enums, config)
+    enum_to_client_attributes = _get_attributes_that_use_enums(enums, config)
     for e in enums:
+        least_restrictive_codegen_method = _get_least_restrictive_codegen_method([
+            config['functions'][f]['codegen_method'] for f in enum_to_client_functions[e]
+        ] + [
+            config['attributes'][a]['codegen_method'] for a in enum_to_client_attributes[e]
+        ])
```

Comment:
You can use a set here instead of a list

Reply:
I have updated it to use `set` instead of `list` here in my latest commit.